unleash-yggdrasil/src/lib.rs

```diff
@@ -303,9 +313,39 @@ impl EngineState {
         }
     }
 
+    fn is_parent_dependency_satisfied(&self, toggle: &CompiledToggle, context: &Context) -> bool {
+        if toggle.dependencies.is_empty() {
+            return true;
+        }
+
+        toggle.dependencies.iter().all(|parent| {
+            let Some(parent_toggle) = self.get_toggle(&parent.feature) else {
+                return false;
+            };
+
+            if !parent_toggle.dependencies.is_empty() {
+                return false;
+            }
+
+            if parent.enabled.unwrap_or(true) {
+                match parent.variants.as_ref() {
+                    Some(variants) if !variants.is_empty() => {
+                        variants.contains(&self.get_variant(&parent.feature, context).name)
+                    }
+                    _ => self.enabled(parent_toggle, context),
+                }
+            } else {
+                !self.enabled(parent_toggle, context)
+            }
+        })
+    }
+
     fn enabled(&self, toggle: &CompiledToggle, context: &Context) -> bool {
-        let context = EnrichedContext::from(context.clone(), toggle.name.clone());
-        toggle.enabled && (toggle.compiled_strategy)(&context)
+        let enriched_context = EnrichedContext::from(context.clone(), toggle.name.clone());
+        match self.is_parent_dependency_satisfied(toggle, context) {
```

Comment:
Matching booleans is weird. This can just be:

```
        toggle.enabled
            && self.is_parent_dependency_satisfied(toggle, context)
            && (toggle.compiled_strategy)(&enriched_context)
```

Reply:
Sure thing!